File: Java/SpaceVIL/spacevil/src/main/java/com/spacevil/Core/KeyArgs.java
package com.spacevil.Core;

public final class KeyArgs extends InputEventArgs
{
    public com.spacevil.Flags.KeyCode key;
    public int scancode;
    public com.spacevil.Flags.InputState state;
    public com.spacevil.Flags.KeyMods mods;

    @Override
    public void clear() {
        key = com.spacevil.Flags.KeyCode.UNKNOWN;
        scancode = -1;
        state = com.spacevil.Flags.InputState.RELEASE;
        mods = com.spacevil.Flags.KeyMods.NO;
    }
}

File: CSharp/SpaceVIL/SpaceVIL/Core/KeyArgs.cs
using System;
using System.Collections.Generic;

namespace SpaceVIL.Core
{
    /// <summary>
    /// A class that describes keyboard key input.
    /// </summary>
    public sealed class KeyArgs : IInputEventArgs
    {
        /// <summary>
        /// Key code as SpaceVIL.Core.KeyCode.
        /// </summary>
        public KeyCode Key;
        /// <summary>
        /// Scancode of key.
        /// </summary>
        public Int32 Scancode;
        /// <summary>
        /// State of input as SpaceVIL.Core.InputState. Values: Release, Press, Repeat.
        /// </summary>
        public InputState State;
        /// <summary>
        /// Used modifiers while typing.
        /// </summary>
        public KeyMods Mods;
        /// <summary>
        /// Clearing KeyArgs.
        /// </summary>
        public void Clear()
        {
            Key = KeyCode.Unknown;
            Scancode = -1;
            State = 0;
            Mods = 0;
        }

        public override String ToString()
        {
            return
                Key + " " +
                Scancode + " " +
                State + " " +
                Mods + " ";
        }
    }
}